Cloud9 integration for Conditions (#210)
aws-samples/cfn101-workshop: content/intermediate/templates/conditions/index.md

Reviewer: Indentation issue in line [286 - 289](https://github.com/aws-samples/cfn101-workshop/pull/210/files#diff-8487daf746d800371c656673b543b8d5a4fb4e18415b40284a4ca07b0b2053f3R286-R289)

When you create the stack, you will pass `test` as the value for `EnvType`, and you will observe only an EC2 instance resource will be provisioned by CloudFormation. Save the template you have updated with content above; next, navigate to the AWS CloudFormation [console](https://console.aws.amazon.com/cloudformation), and choose to create a stack using this template:

:::::tabs{variant="container"}
::::tab{id="cloud9" label="Cloud9"}
1. In the **Cloud9 terminal** navigate to `code/workspace/conditions`:
:::code{language=shell showLineNumbers=false showCopyAction=true}
cd cfn101-workshop/code/workspace/conditions
:::

> 1. Use the AWS CLI to create the stack. The required parameter `--template-body` have been pre-filled for you.

Reviewer 2: ```suggestion
1. Use the AWS CLI to create the stack. The required parameters have been pre-filled for you.
```

:::code{language=shell showLineNumbers=false showCopyAction=true}

> aws cloudformation create-stack --stack-name cfn-workshop-condition-test --template-body file://condition-resource.yaml --parameters ParameterKey="EnvType",ParameterValue="test"

Reviewer 2: ```suggestion
aws cloudformation create-stack --stack-name cfn-workshop-condition-test \
--template-body file://condition-resource.yaml \
--parameters ParameterKey="EnvType",ParameterValue="test"
```

:::
1. If the `create-stack` command was successfully sent, CloudFormation will return `StackId`.
:::code{language=shell showLineNumbers=false showCopyAction=false}
"StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/cfn-workshop-condition-test/739fafa0-e4d7-11ed-a000-12d9009553ff"
:::
1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** console in a new tab and check if the stack status is **CREATE_COMPLETE**.
::::
::::tab{id="local" label="Local development"}
1. In the CloudFormation console, select **Create stack**, **With new resources (standard)**.
1. In **Prepare template**, select **Template is ready**.
1. In **Template source**, select **Upload a template file**.
1. Choose the `condition-resource.yaml` template.
1. Enter a **Stack name**. For example, choose to specify `cfn-workshop-condition-test`.
1. Pass `test` as the value for the `EnvType` parameter. Choose **Next**.
1. Choose to accept default values in the **Configure stack options** page. Choose **Next**.

> 1. Choose **Create stack**. You can view the progress of the stack being created in the CloudFormation console.

Reviewer: ```suggestion
1. Choose **Submit**. You can view the progress of the stack being created in the CloudFormation console.
```

::::
:::::
Once the stack is in the `CREATE_COMPLETE` status, navigate to the **Resources** tab for your stack: verify the only resource provisioned is your EC2 instance, based on the logic you created driven by the `test` value you passed to the `EnvType` parameter, and to the condition you added and associated to the other two resources in the template:

![condition-test](/static/intermediate/templates/conditions/condition-test.png)

In the next step, you will create a new stack with the same template. This time, you will pass `prod` as the value for the `EnvType` parameter, and verify that you will provision, with CloudFormation, your `Volume` and `MountPoint` resources as well. Navigate to the AWS CloudFormation [console](https://console.aws.amazon.com/cloudformation), and choose to create a stack using your existing template:

:::::tabs{variant="container"}
::::tab{id="cloud9" label="Cloud9"}
1. In the **Cloud9 terminal** navigate to `code/workspace/conditions`:
:::code{language=shell showLineNumbers=false showCopyAction=true}
cd cfn101-workshop/code/workspace/conditions
:::

> 1. Use the AWS CLI to create the stack. The required parameter `--template-body` have been pre-filled for you.

Reviewer 2: ```suggestion
1. Use the AWS CLI to create the stack. The required parameters have been pre-filled for you.
```

:::code{language=shell showLineNumbers=false showCopyAction=true}

> aws cloudformation create-stack --stack-name cfn-workshop-condition-prod --template-body file://condition-resource.yaml --parameters ParameterKey="EnvType",ParameterValue="prod"

Reviewer 2: ```suggestion
aws cloudformation create-stack --stack-name cfn-workshop-condition-prod \
--template-body file://condition-resource.yaml \
--parameters ParameterKey="EnvType",ParameterValue="prod"
```

When you create the stack, you will pass `test` as the value for `EnvType`, and you will observe only an EC2 instance resource will be provisioned by CloudFormation. Save the template you have updated with content above; next, navigate to the AWS CloudFormation [console](https://console.aws.amazon.com/cloudformation), and choose to create a stack using this template:

:::::tabs{variant="container"}
::::tab{id="cloud9" label="Cloud9"}
1. Upload the file to your **template S3 bucket** using AWS CLI [aws s3 cp](https://docs.aws.amazon.com/cli/latest/reference/s3/cp.html) command.

>    `aws s3 cp code/workspace/condition-resource.yaml s3://cfn-workshop-01-{accountid}`

Reviewer 2: Can we please use code blocks to make it clearer and neater?
`:::code{language=shell showLineNumbers=false showCopyAction=true}
aws s3 cp code/workspace/condition-resource.yaml s3://cfn-workshop-01-{accountid}
:::`

>    `aws s3 cp code/workspace/condition-resource.yaml s3://cfn-workshop-01-{accountid}`

Reviewer 2: ```suggestion
:::code{language=shell showLineNumbers=false showCopyAction=true}
aws s3 cp code/workspace/condition-resource.yaml s3://cfn-workshop-01-{accountid}
:::
```

1. Determine the **Object URL** as you'll need it in the next step, based on this format `https://[bucketname].s3.amazonaws.com/[key]`
    for example:-

>      `https://cfn-workshop-01-{accountid}.s3.amazonaws.com/condition-resource.yaml`

Reviewer 2: comment block
`:::code{language=shell showLineNumbers=false showCopyAction=true}
https://cfn-workshop-01-{accountid}.s3.amazonaws.com/{template-name}.yaml
:::`

>      `https://cfn-workshop-01-{accountid}.s3.amazonaws.com/condition-resource.yaml`

Reviewer 2: ```suggestion
:::code{language=shell showLineNumbers=false showCopyAction=true}
https://cfn-workshop-01-{accountid}.s3.amazonaws.com/condition-resource.yaml
:::
```

1. Wait until the stack creation is complete. Refresh the view in the console until you see your stack to be in the `CREATE_COMPLETE` status.
::::
:::::
Once the stack is in the `CREATE_COMPLETE` status, navigate to the **Resources** tab for your stack: verify the only resource provisioned is your EC2 instance, based on the logic you created driven by the `test` value you passed to the `EnvType` parameter, and to the condition you added and associated to the other two resources in the template:

![condition-test](/static/intermediate/templates/conditions/condition-test.png)

In the next step, you will create a new stack with the same template. This time, you will pass `prod` as the value for the `EnvType` parameter, and verify that you will provision, with CloudFormation, your `Volume` and `MountPoint` resources as well. Navigate to the AWS CloudFormation [console](https://console.aws.amazon.com/cloudformation), and choose to create a stack using your existing template:

:::::tabs{variant="container"}
::::tab{id="cloud9" label="Cloud9"}
1. Upload the file to your **template S3 bucket** using AWS CLI [aws s3 cp](https://docs.aws.amazon.com/cli/latest/reference/s3/cp.html) command.

>    `aws s3 cp code/workspace/condition-resource.yaml s3://cfn-workshop-01-{accountid}`

Reviewer 2: comment block

In this section, you will pass `test` as the value for the `EnvType` parameter, and verify the type of your EC2 instance will be `t2.micro`. Navigate to the AWS CloudFormation [console](https://console.aws.amazon.com/cloudformation), and choose to create a stack using this template:

:::::tabs{variant="container"}
::::tab{id="cloud9" label="Cloud9"}
1. In the **Cloud9 terminal** navigate to `code/workspace/conditions`:
:::code{language=shell showLineNumbers=false showCopyAction=true}
cd cfn101-workshop/code/workspace/conditions
:::

> 1. Use the AWS CLI to create the stack. The required parameter `--template-body` have been pre-filled for you.

Reviewer 2: ```suggestion
1. Use the AWS CLI to create the stack. The required parameters have been pre-filled for you.
```

:::code{language=shell showLineNumbers=false showCopyAction=true}

> aws cloudformation create-stack --stack-name cfn-workshop-condition-property-test --template-body file://condition-resource-property.yaml --parameters ParameterKey="EnvType",ParameterValue="test"

Reviewer 2: ```suggestion
aws cloudformation create-stack --stack-name cfn-workshop-condition-property-test \
--template-body file://condition-resource-property.yaml \
--parameters ParameterKey="EnvType",ParameterValue="test"
```

:::
1. If the `create-stack` command was successfully sent, CloudFormation will return `StackId`.
:::code{language=shell showLineNumbers=false showCopyAction=false}
"StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/cfn-workshop-condition-property-test/739fafa0-e4d7-11ed-a000-12d9009553ff"
:::
1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** console in a new tab and check if the stack status is **CREATE_COMPLETE**.
::::
::::tab{id="local" label="Local development"}
1. In the CloudFormation console, select **Create stack**, **With new resources (standard)**.
1. In **Prepare template**, select **Template is ready**.
1. In **Template source**, select **Upload a template file**.
1. Choose the `condition-resource-property.yaml` template.
1. Enter a **Stack name**. For example, choose to specify `cfn-workshop-condition-property-test`.
1. Pass `test` as the value for the `EnvType` parameter. Choose **Next**.
1. Choose to accept default values in the **Configure stack options** page. Choose **Next**.

> 1. Choose **Create stack**. You can view the progress of the stack being created in the CloudFormation console.

Reviewer 2: ```suggestion
1. Choose **Submit**. You can view the progress of the stack being created in the CloudFormation console.
```

In this section, you will pass `test` as the value for the `EnvType` parameter, and verify the type of your EC2 instance will be `t2.micro`. Navigate to the AWS CloudFormation [console](https://console.aws.amazon.com/cloudformation), and choose to create a stack using this template:

:::::tabs{variant="container"}
::::tab{id="cloud9" label="Cloud9"}
1. Upload the file to your **template S3 bucket** using AWS CLI [aws s3 cp](https://docs.aws.amazon.com/cli/latest/reference/s3/cp.html) command.

>    `aws s3 cp code/workspace/condition-resource.yaml s3://cfn-workshop-01-{accountid}`

Reviewer 2: comment block

1. Determine the **Object URL** as you'll need it in the next step, based on this format `https://[bucketname].s3.amazonaws.com/[key]`
    for example:-

>      `https://cfn-workshop-01-{accountid}.s3.amazonaws.com/condition-resource.yaml`

Reviewer 2: comment block

>      `https://cfn-workshop-01-{accountid}.s3.amazonaws.com/condition-resource.yaml`

Reviewer 2: ```suggestion
:::code{language=shell showLineNumbers=false showCopyAction=true}
https://cfn-workshop-01-{accountid}.s3.amazonaws.com/condition-resource.yaml
:::
```

Next, navigate to the AWS CloudFormation [console](https://console.aws.amazon.com/cloudformation), and choose to  update your `cfn-workshop-condition-prod` stack:

:::::tabs{variant="container"}
::::tab{id="cloud9" label="Cloud9"}
1. In the **Cloud9 terminal** navigate to `code/workspace/conditions`:
  :::code{language=shell showLineNumbers=false showCopyAction=true}
  cd cfn101-workshop/code/workspace/conditions
  :::

> 1. Use the AWS CLI to update the stack. The required parameter `--template-body` have been pre-filled for you.

Reviewer 2: ```suggestion
1. Use the AWS CLI to update the stack. The required parameters have been pre-filled for you.
```

  :::code{language=shell showLineNumbers=false showCopyAction=true}

>   aws cloudformation update-stack --stack-name cfn-workshop-condition-prod --template-body file://condition-resource.yaml --parameters ParameterKey="EnvType",ParameterValue="prod"

Reviewer 2: ```suggestion
  aws cloudformation update-stack --stack-name cfn-workshop-condition-prod \
  --template-body file://condition-resource.yaml \
  --parameters ParameterKey="EnvType",ParameterValue="prod"
```

  :::
1. If the `update-stack` command was successfully sent, CloudFormation will return `StackId`.
  :::code{language=shell showLineNumbers=false showCopyAction=false}

>   "StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/cfn-workshop-prod/739fafa0-e4d7-11ed-a000-12d9009553ff"

Reviewer: ```suggestion
  "StackId": "arn:aws:cloudformation:us-east-1:123456789012:stack/cfn-workshop-condition-prod/739fafa0-e4d7-11ed-a000-12d9009553ff"
```

1. Open the **[AWS CloudFormation](https://console.aws.amazon.com/cloudformation)** console in a new tab and check if the stack status is **UPDATE_COMPLETE**.
::::
::::tab{id="local" label="Local development"}
1. In the CloudFormation console, select **Update stack**.
1. In **Prepare template**, select **Replace current template**.
1. In **Template source**, select **Upload a template file**.
1. Choose the `condition-resource.yaml` template.
1. `EnvType` should already be set to `prod`. Choose **Next**.
1. Choose to accept default values in the **Configure stack options** page. Choose **Next**.

> 1. Choose **Update stack**. You can view the progress of the stack being created in the CloudFormation console.

Reviewer: ```suggestion
1. Choose **Submit**. You can view the progress of the stack being created in the CloudFormation console.
```

Next, navigate to the AWS CloudFormation [console](https://console.aws.amazon.com/cloudformation), and choose to  update your `cfn-workshop-condition-prod` stack:

  :::::tabs{variant="container"}
    ::::tab{id="cloud9" label="Cloud9"}
    1. Upload the file to your **template S3 bucket** using AWS CLI [aws s3 cp](https://docs.aws.amazon.com/cli/latest/reference/s3/cp.html) command.

>         `aws s3 cp code/workspace/condition-resource.yaml s3://cfn-workshop-01-{accountid}`

Reviewer 2: comment block

    1. Determine the **Object URL** as you'll need it in the next step, based on this format `https://[bucketname].s3.amazonaws.com/[key]`
        for example:-

>         `https://cfn-workshop-01-{accountid}.s3.amazonaws.com/condition-resource.yaml`

Reviewer 2: comment block

>         `https://cfn-workshop-01-{accountid}.s3.amazonaws.com/condition-resource.yaml`

Reviewer 2: ```suggestion
:::code{language=shell showLineNumbers=false showCopyAction=true}
https://cfn-workshop-01-{accountid}.s3.amazonaws.com/condition-resource.yaml
:::
```
